daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -358,23 +357,10 @@ final class ImmArray[+A] private (
 }
 
 object ImmArray extends ImmArrayInstances {
-  def empty[T]: ImmArray[T] =
-    ImmArray.fromArraySeq(ArraySeq.empty[AnyRef].asInstanceOf[ArraySeq[T]])
 
-  def apply[T](element0: T, elements: T*): ImmArray[T] = {
-    val builder = ImmArray.newBuilder[T]
-    builder += element0
-    builder ++= elements
-    builder.result()
-  }
+  private[this] val Empty = ImmArray.fromArraySeq(ArraySeq.empty[AnyRef])
```

Comment:
I'm _fairly_ certain this should work; Scala sometimes requires you to pass `Nothing` as an explicit tparam even when it's the only possible choice, so might need to be more verbose.

```suggestion
  private[this] val Empty: ImmArray[Nothing] = ImmArray.fromArraySeq(ArraySeq.empty)
```